include/oneapi/dpl/internal/serial_numeric_impl.h

```diff
@@ -0,0 +1,49 @@
+/*
+ *  Copyright (c) Intel Corporation
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+#ifndef _ONEDPL_SERIAL_NUMERIC_IMPL_H
+#define _ONEDPL_SERIAL_NUMERIC_IMPL_H
+
+#include <iterator>
+namespace oneapi
```

Comment:
Hi, @capatober 
I tried another way to solve the conflicts. We always provide our own impl for serial reduce in oneapi::dpl. However, this solution still depends on order. Our own impl must be after injecting oneapi::dpl::reduce into std namespace otherwise it will conflict with libstdc++'s serial reduce.
Thanks very much.

Reply:
Ge Jin, have you checked whether the proposed implementation does not depend on the order of including `<numeric>` and `<execution>`, i.e. works correctly with any order of the two? That's not clear to me from the discussion; please confirm.

Reply:
Hi, @akukanov 
We have already had test cases for testing inclusion order of <numeric> and <execution>:
https://github.com/oneapi-src/oneDPL/blob/main/test/general/header_inclusion_order_numeric_0.pass.cpp
https://github.com/oneapi-src/oneDPL/blob/main/test/general/header_inclusion_order_numeric_1.pass.cpp

Are these 2 tests what you want?
Thanks very much.

Reply:
Yes, that's enough. Thanks.